### PR TITLE
[show] add new CLI to show tunnel route objects

### PR DESCRIPTION
**Pull request by zjswhhh**
#### What I did
Add new CLI support to show tunnel route objects in ASIC DB.

sign-off: Jing Zhang zhangjing@microsoft.com

#### How I did it
* Check if tunnel route object exists for **server_ipv4, server_ipv6, soc_ipv4**. If existing, print it out. 
* If not specifying port name, print all tunnel route objects. 
  
#### How to verify it
* Added unit tests. 
* Tested on dual testbeds: 
  *  show mux tunnel-route Ethernet140
     ```
      PORT         DEST_TYPE    DEST_ADDRESS    
      -----------  -----------  ---------------  
      Ethernet140  server_ipv4  192.168.0.29/32  
     ```
  *  show mux tunnel-route --json Ethernet140
     ```
      {
          "TUNNEL_ROUTE": {
              "Ethernet140": {
                  "server_ipv4": {
                      "DEST": "192.168.0.29/32"
                  }
              }
          }
      }
     ```

### CODE REVIEW

**Comment by zjswhhh** [comment]
> I tent to disagree with this change. First of all, `Nexthop ID` doesn't convey any meaningful information. Secondly, just "ip route show 192.168.0.29/32" will display if its tunnel or non-tunnel route.

Hi Prince - Yes `ip route show` will have that information. It's just recently we have been doing static active-active (forcing both ToR to be active) on active-standby dualToRs, and to verify the tunnel route is removed, I first need to get server ip and then check the route.  So I wanted to add a CLI to check in batches. But it's not a must-have.

**Review by yxieca — DISMISSED** [review]

File: show/muxcable.py [hunk 1]
```diff
@@ -503,6 +503,42 @@ def create_json_dump_per_port_config(db, port_status_dict, per_npu_configdb, asi
     if soc_ipv4_value is not None:
         port_status_dict["MUX_CABLE"]["PORTS"][port_name]["SERVER"]["soc_ipv4"] = soc_ipv4_value
 
+def get_tunnel_route_per_port(db, port_tunnel_route, per_npu_configdb, per_npu_asic_db, asic_id, port):
```
Comment (prsunny) [inline show/muxcable.py:506]: If the intention is to only display the list of Tunnel route, please fetch data from the APP_DB "TUNNEL_ROUTE_TABLE" and display and avoid this complexity of looping through ASIC DB.

**Review by prsunny — CHANGES_REQUESTED** [review]

File: show/muxcable.py (hunk 1 above)
Comment (zjswhhh) [reply show/muxcable.py:506]: Will there be any chance of inconsistency between APP_DB and ASIC_DB? Are there any concerns of using ASIC_DB?

Comment (prsunny) [reply show/muxcable.py:506]: There should not be any inconsistency. If it is there, its a bug scenario. You don't have to use the translation logic  for ASIC_DB objects if using APP_DB. Moreover, its straightforward. Tunnel routes is what you are looking for and it is part of the TUNNEL_ROUTE_TABLE.

Comment (zjswhhh) [reply show/muxcable.py:506]: Thanks for clarifying. Will update accordingly.

**Review by prsunny — APPROVED** [review]
lgtm

**Review by yxieca — APPROVED** [review]

**Comment by qiluo-msft** [comment]
This commit could not be cleanly cherry-picked to 202012. Please submit another PR.

**Comment by liuh-80** [comment]
Cherry-pick PR merged: https://github.com/sonic-net/sonic-utilities/pull/2659

**Pull request merged**